### ISSUE TITLE
Is Firefox a second-class platform for Vimium?

### ISSUE THREAD

**Issue by finalfantasia** [philc/vimium#2985]
Hi!

Thank you very much for creating this awesome extension and porting it to Firefox! I love it!

I've noticed a few issues that gave me an impression that Firefox is a second-class platform for Vimium:

1. It is called "Vimium-FF" on AMO but shows up as "Vimium" in Add-ons Manager in Firefox after installation. Was there any reason you could not use the original name (i.e., "Vimium") on AMO? I noticed that it was initially called "Vimium-UX" and then later renamed to "Vimium-FF", what was the reason for this change?

2. It has a logo of Chromium as a background in its own logo. Seeing a Chromium logo in Firefox feels... just weird. Is there any reason it cannot use a new, universal logo in both browsers? The browser action icon (i.e., a styled "V" on a light blue background that shows up on the toolbar) looks like the perfect candidate for this. It looks simple, modern, and flat. Is there anything planned for this?

3. "README.md" uses Google Chrome to describe Vimium's features and briefly mentions Firefox near the end of it which is almost not noticeable. Is there any reason for this? Is there any way we can help to "neutralize" it to make it not browser-specific?

4. It's been in the "experimental" state for a while, what's holding it back from being published as a non-experimental extension? It seems to be working just perfectly fine for me.

There might be other issues that I've not noticed yet. Is there any way I, as a Firefox user, can help to improve the overall experience of Vimium on Firefox?

Thanks!

**Comment by innaterebel** [comment]
I will go ahead and have things that might be a bit embarrassing for the developer to say be spoken.

Vimium is named after Chromium, apparently. Developed for Chrome long before Firefox moved to WebExtensions. Naturally Chrome is the native platform for Vimium, everything are illustrated based on Chrome. The Firefox version was ported due to high demand after Firefox's departure from legacy add-ons as one of the most suitable VimFX alternatives.

Vimium, after years of development, have built a sensible set of function at least for Chrome. 
It currently **has only one or two active developer**, and they don't use Firefox that much, if at all. They are not super familiar with Firefox's design, **and only have so much time**. To be expected they have things overlooked here and there now and then, especially non-functional non-critical bits; they also know it well that they keep the experimental state for Firefox.

They are doing their best, technical-wise and time-wise. Everybody is welcomed to help completing the experience of Vimium, good for digging bugs out, better for pull requests, ad stated by README.md.

**Comment by smblott-github** [comment]
> It is called "Vimium-FF"...

There's a very-old (2012), independent [Vimium port](https://addons.mozilla.org/en-GB/firefox/addon/vimium/) sitting on the name "Vimium".

**Comment by finalfantasia** [comment]
Hi @innaterebel and @smblott-github , thank you for your replies.

I had been a long-time Vimium (Google Chrome version) user before I became a Firefox convert. Since then, I had been a happy user of VimFx up until the WebExtension fiasco. After that, I stopped using anything that emulates Vim in Firefox (call it being lazy or anything else). But recently, I started looking for a keyboard-friendly extension that shows a searchable list of tabs when pressing a certain kind of key combination (much like what IntelliJ IDEA does when you hit **⌘E** (Recent Files)) but could not find any. I thought of writing one that does exactly that, no more or less. But before doing that, for some reason, I remembered Vimium-FF and gave it a shot. To my surprise, that feature is already in it (**T** (Search through your open tabs))! So I thought to myself, I would've gone through the hassle and reinvented the wheels, in an inferior way, had I not tried Vimium-FF, why not contributing that time and energy to something that's already this awesome? And that led us to this issue (with an intentionally click-baity title, my apologies **:-)**)

So the point is, I'm here to help. I've been thinking of giving back to the open-source community and have just started dipping my toe nails in the water. I think I can contribute at least an hour or two weekly and thus wanted to start small just to have a feeling of it. I can, of course, submit a bunch of PRs to "fix" those issues that I listed in my first comment but they are in the surface areas that are pretty critical for an extension (name, logo, README.md, etc.) so I didn't want to take it lightly and end up with a bunch of rejected PRs, wasting our time and energy and, probably, kill my own desire of giving back to the community right int the cradle :-))

Now onto the name of the extension.

Running into a name conflict seems to be common on AMO and there seems to be a way [0] to resolve this with the help of AMO administrators. I looked at that old port of Vimium from 2012 and noticed that it had not been updated ever since then. And the author had clearly stated that he didn't intend to maintain that code base and instead had recommended VimFx to his potential users [1]. It would be a pity if a good name, such as Vimium, died with an abandoned extension. My question for @smblott-github is, have you contacted any of the AMO administrators to see if there's any possibility to reclaim that name so that it can be used for this modern port of Vimium? If not, I am willing to talk the talk and walk the walk on your behalf (not sure how it works yet, but giving it a try won't harm). For the rest of the "easy" issues, I can, in fact, start working on PRs as long as I can get a green light from the maintainers.

Thanks!

[0] https://discourse.mozilla.org/t/not-able-to-submit-new-add-on-name-taken/25595
[1] https://code.google.com/archive/p/vimium-firefox/
